pkg/autohealing/cloudprovider/openstack/provider.go

```diff
@@ -168,13 +193,21 @@ func (provider OpenStackCloudProvider) Repair(nodes []healthcheck.NodeInfo) erro
 	if len(nodes) == 0 {
 		return nil
 	}
-
 	clusterName := provider.Config.ClusterName
+	err := provider.UpdateHealthStatus(nodes)
+	if err != nil {
+		return fmt.Errorf("Failed to update the helath status of cluster %s, error: %v", clusterName, err)
+	}
+
 	cluster, err := clusters.Get(provider.Magnum, clusterName).Extract()
 	if err != nil {
 		return fmt.Errorf("failed to get the cluster %s, error: %v", clusterName, err)
 	}
 
+	if err := provider.waitForClusterComplete(cluster.UUID, 30*time.Second); err != nil {
```

Comment:
Why we need to call this function here? Shouldn't it be part of checks in `Enabled()` function?

Reply:
Yep, `Enabled()` could be a better place. But no matter we put the wait here or in Enabled(), we do need to have this check because the health status update won't trigger the Heat stack update, so the Heat statck status won't be changed. That means the existing stack status check in Enabled is not enough.

I will move the check to Enabled().